util/util.go

```diff
@@ -69,6 +62,9 @@ func continueWithError(err error, msg string) string {
 }
 
 func CleanIDs(list string) []string {
+	if list == "" {
+		return nil
+	}
```

Comment:
Is this necessary? I think the subsequent code will handle it regardless (and less code is easier to read)